plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/OpenShiftResourceUIModel.java

```diff
@@ -19,30 +21,41 @@
  * Display model for OpenShift resources to allow
  * overriding equals and hash
  * @author jeff.cantrill
+ * @author Jeff Maury
  *
  */
 public class OpenShiftResourceUIModel extends ObservableUIPojo implements IResourceUIModel{
 
-	private final IResource resource;
-	private final String tostring;
+	private IResource resource;
+	private String tostring;
 	private final Object parent;
+	private AtomicBoolean deleting = new AtomicBoolean(false);
+
 
-	public OpenShiftResourceUIModel(IResource resource, IResourcesUIModel parent){
+	public OpenShiftResourceUIModel(IResource resource, Object parent){
```

Comment:
This is a step back from the typed model. We have 
	ResourcesUIModel (parent can be defined with a type parameter), 
	Deployment (parent IProjectAdapter), 
	OpenShiftProjectUIModel (parent IOpenShiftConnection), 
	OpenShiftResourceUIModel (parent is IResourcesUIModel)
Is it necessary for ResourcesUIModel to extend OpenShiftResourceUIModel, or rather we can implement IResourceUIModel?
In each class and where possible, in interfaces, method getParent() may be overridden to with correct parent return type.
As a maximum, IAncestorable should declare a type parameter for the parent.